x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -45,4 +59,85 @@ public LogicalPlan optimize(LogicalPlan verified) {
 
         return Arrays.asList(operators, label);
     }
+
+
+    private static class ReplaceWildcards extends OptimizerRule<Filter> {
+
+        private static boolean isWildcard(Expression expr) {
+            if (expr instanceof Literal) {
+                Literal l = (Literal) expr;
+                if (l.value() instanceof String) {
```

Comment:
Better to check if the expression is foldable and its folded value contains ("*"):

```java
if (exp.foldable()) {
   Object value = exp.fold();
   return value instanceof String && value.toString().contains("*");
}
return false;
```